21 crear tests para el modelo de book

paul-ot opened the pull request:
Trello Card: https://trello.com/c/atYKD3h4/21-crear-tests-para-el-modelo-de-book

Installed rspec, factory_bot_rails and shoulda-matchers gems. I've also created factories and small tests for both User and Book models.

paul-ot committed:
Added Rspec gem

paul-ot committed:
installed rspec, factory_bot_rails and shoulda-matchers gems. I've also created factories and small tests for both User and Book models.

holywyvern commented on spec/factories/books.rb:
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :book do
+    gender { Faker::Name.name }
```
Mejores opciones hay en `Faker::Book` https://github.com/stympy/faker/blob/master/doc/v1.9.1/book.md tambien el genero de un libro en inglés es genre, no gender.

holywyvern replied:
Quizas es mejor que agregues una migracion con `rename_column :book, :gender, :genre`

paul-ot replied:
Solucionado.

holywyvern commented on spec/factories/users.rb:
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :user do
+    email         { Faker::Internet.email}
+    password      { 'test911' }
```
Para Devise es necesario que tambien agregues `password_confirmation: { password }` o fallan los create

paul-ot replied:
buen punto.

paul-ot replied:
Solucionado.

holywyvern commented on spec/spec_helper.rb:
```diff
@@ -0,0 +1,97 @@
+  config.filter_run_when_matching :focus
+
+  # Allows RSpec to persist some state between runs in order to support
+  # the `--only-failures` and `--next-failure` CLI options. We recommend
```
Recorda quitar el código comentado

paul-ot replied:
Solucionado.

paul-ot committed:
Fixed issues:
- Better fake test for book fields
- Added password_confirmation to the Users factory.
- Removed comments from spec_helper and rails_helper
- Renamed field books.genere to books.genre.
